vcr/holder/interface.go

```diff
@@ -34,10 +35,18 @@ var VerifiablePresentationLDType = ssi.MustParseURI("VerifiablePresentation")
 
 // Wallet holds Verifiable Credentials and can present them.
 type Wallet interface {
+	Diagnostics() []core.DiagnosticResult
```

Comment:
I did that for the `OpenIDHandler`s, but it leads to it own set of complexity, notably having to maintain a map of `Wallet` instances in memory, since we now cache diagnostics (thus have state). And since wallets are sort of dynamic, because DIDs can be created/deactivated on the fly, this would lead to lots of complicated code to have a more domain-driven API interface. So I think I prefer to have a holder-agnostic interface in which you supply the holder DID when required?

Also complicates `Put` and `BuildPresentation`; what if you pass a VC you don't own?